### ISSUE TITLE
noSpecials for propto

### ISSUE THREAD

**Issue by mmcgillycuddy** [bbolker/reformulas#2]
Hi Ben,

I'm not sure if this is the problem, but I am getting a problem with the formula in glmmTMB. Let me know if this should be in the glmmTMB issues. I think when I have a propto structure in the formula it doesn't get rid of propto with the `noSpecials` function.

```
> noSpecials(sub_specials(y ~ x + homdiag(1 + x | spp)), delete = F)
y ~ x + (1 + x + spp)
```
```
> noSpecials(sub_specials(y ~ x + propto(1 + x | spp)), delete = F)
y ~ x + propto(1 + x + spp)
```

**Comment by bbolker** [comment]
Yes, the problem is [here](https://github.com/bbolker/reformulas/blob/0080169895a40124232b02c67093cca0f70ca422/R/utils.R#L199-L219); there's a hard-coded list of specials. I had already flagged this as a potential issue; have to think about how to pass the information between packages (without having to update in both places), and without some incredibly tedious process of passing information down through a stack of function calls ...

**Comment by mmcgillycuddy** [comment]
Yes, that worked. Thanks Ben.

**Comment by bbolker** [comment]
Arguably `findReTrmClasses()` shouldn't use a hard-coded list at all ...?

**Comment by mebrooks** [comment]
Sorry for the delay. I merged the pull request that makes it possible to pass known covstructs to reformulas.
